src/app/services/send-page-data/send-page-data.ts

```diff
@@ -0,0 +1,211 @@
+      this.dataSubject.next(this.dataBundle)
+      return //validation errors, so return early
+    }
+
+    this.getEstimatedExchange(originCoin, 'nav', transferAmount)
+      .then((data) => {
+        if(originCoin === 'nav'){
+          this.dataBundle.changellyFeeTotalToNav = 0
+        } else {
+          this.dataBundle.changellyFeeTotalToNav = transferAmount - transferAmount * this.CHANGELLY_FEE
+        }
+        this.dataBundle.estConvToNav = this.dataBundle.changellyFeeTotalToNav
+        this.dataBundle.navtechFeeTotal = this.dataBundle.estConvToNav - this.dataBundle.estConvToNav * this.NAVTECH_FEE
+        this.dataBundle.estNavAfterNavtechFee = this.dataBundle.estConvToNav - this.dataBundle.changellyFeeTotalToNav
+
+        this.getEstimatedExchange('nav', destCoin, this.dataBundle.estNavAfterNavtechFee)
+        .then((data) => {
+          if(destCoin === 'nav'){
+            this.dataBundle.changellyFeeTotalFromNav = 0
+          } else {
+            this.dataBundle.changellyFeeTotalFromNav = this.dataBundle.estNavAfterNavtechFee - this.dataBundle.estNavAfterNavtechFee * this.CHANGELLY_FEE
+          }
+          this.dataBundle.estConvFromNav = data
+          this.isDataSet = true
+
+          this.validateDataBundle(this.dataBundle)
+          .then(() => this.dataSubject.next(this.dataBundle))
+        })
+    })
+  }
+
+  validateFormData(dataBundle):boolean {
+    let validData = true
+    if(!Number.isInteger(dataBundle.transferAmount)){
+      dataBundle.errors.invalidTransferAmount = true
+      dataBundle.validData = false
+      validData = false
+    }
+    if(dataBundle.originCoin === 'nav' && dataBundle.destCoin === 'nav') {
+      dataBundle.errors.navToNavTransfer = true
+      dataBundle.validData = false
+      validData = false
+    }
+    if(dataBundle.transferAmount < this.getMinTransferAmount(dataBundle.originCoin, 'nav')) {
+      dataBundle.errors.transferTooSmall = true
+      dataBundle.validData = false
+      validData = false
+    }
+    return validData
+  }
+
+  validateDataBundle(dataBundle) {
+    return new Promise<any>( resolve => {
+      if(dataBundle.estNavAfterNavtechFee > this.MAX_NAV_PER_TRADE) {
+        dataBundle.errors.transferTooLarge = true
+        dataBundle.validData = false
+      }
+      if(!this.checkAddressIsValid(dataBundle.destAddr)) {
+        dataBundle.errors.invalidDestAddress  = true
+        dataBundle.validData = false
+      }
+      // if(changellyError () {
+        // dataBundle.errors.changellyError = true
+      // }
+      resolve()
+    })
+  }
+
+  resetDataBundleErrors(bundle) {
+    bundle.validData = true
+    bundle.errors.invalidDestAddress = false
+    bundle.errors.invalidTransferAmount = false
+    bundle.errors.transferTooSmall = false
+    bundle.errors.transferTooLarge = false
+    bundle.errors.navToNavTransfer = false
+    bundle.errors.changellyError = false
+  }
+
+  checkAddressIsValid(address) {
+    return address ?  true : false
```

Comment:
what is this checking? do we get an invalid address response from changelly? or do we need to implement something ourselves?